File: src/Comun/ObjetosComunes/ContratosServicios/IServicioBase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Trascend.Bolet.ObjetosComunes.ContratosServicios
{
    public interface IServicioBase<Entidad>
    {
        IList<Entidad> ConsultarTodos();

        Entidad ConsultarPorId(Entidad entidad);

        bool InsertarOModificar(Entidad entidad, int hash);

        bool Eliminar(Entidad entidad, int hash);

        bool VerificarExistencia(Entidad entidad);
    }
}

File: trunk/src/Comun/ObjetosComunes_Fac/ContratosServicios/IServicioBase.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace ContratosServicios
    Public Interface IServicioBase(Of Entidad)
        Function ConsultarTodos() As IList(Of Entidad)

        Function ConsultarPorId(ByVal entidad As Entidad) As Entidad

        Function InsertarOModificar(ByVal entidad As Entidad, ByVal hash As Integer) As Boolean

        Function Eliminar(ByVal entidad As Entidad, ByVal hash As Integer) As Boolean

        Function VerificarExistencia(ByVal entidad As Entidad) As Boolean
    End Interface
End Namespace